src/app/hooks/useRequest.ts

```diff
@@ -5,15 +5,15 @@ import { Nullable } from '~utils/types';
 
 export type Error<E> = { problem: PROBLEM_CODE; errorData?: E };
 type Request<P, D, E> = (params: P) => Promise<ApiResponse<D, E>>;
-type Success<D> = (data?: ApiResponse<D>) => void;
+type Success<D> = (data?: ApiOkResponse<D>) => void;
 type Failure<E> = (error: Error<E>) => void;
 type PostFetch<D, E> = (response: ApiOkResponse<D> | ApiErrorResponse<E>) => void;
 
 interface AsyncRequestHookParams<P, D, E> {
   request: Request<P, D, E>;
   withPostSuccess?: Success<D>;
   withPostFailure?: Failure<E>;
-  initialState?: D | null;
+  initialState?: ApiOkResponse<D> | null;
```

Comment:
this is wrong, initialState should be `D` only, if not you have to do this: `{ ok: true, data: ...yourResponse }`